packages/scaffold-ui/src/views/w3m-swap-view/index.ts

```diff
@@ -185,21 +185,24 @@ export class W3mSwapView extends LitElement {
     )
     const amount = target === 'toToken' ? this.toTokenAmount : this.sourceTokenAmount
     const price = target === 'toToken' ? this.toTokenPriceInUSD : this.sourceTokenPriceInUSD
-    let value = parseFloat(amount) * price
+    const marketValue = NumberUtil.parseLocalStringToNumber(amount) * price
 
-    if (target === 'toToken') {
-      value -= this.gasPriceInUSD || 0
-    }
+    // Why do we need to subtract the gas price here?
+    /*
+     * If (target === 'toToken') {
+     *   marketValue -= this.gasPriceInUSD || 0
+     * }
+     */
```

Comment:
probably because we are showing this value to the user as what it'd receive
given they are spending gas for the tx, it's deducted

buyt cc @enesozturk @svenvoskamp for confirmation